mock/mock.go

```diff
@@ -296,7 +296,15 @@ func (m *Mock) Called(arguments ...interface{}) Arguments {
 		closestFound, closestCall := m.findClosestCall(functionName, arguments...)
 
 		if closestFound {
-			panic(fmt.Sprintf("\n\nmock: Unexpected Method Call\n-----------------------------\n\n%s\n\nThe closest call I have is: \n\n%s\n", callString(functionName, arguments, true), callString(functionName, closestCall.Arguments, true)))
+			if functionName == closestCall.Method && closestCall.Repeatability == -1 {
+				panic(fmt.Sprintf(
+					"\n\nmock: Unexpected Method Call\n-----------------------------\n\n%s\n\n" +
+					"Found call, but Repeatability is exhausted. Use Times(n), Once(), Twice() or nothing to finetune.\n\n",
```

Comment:
That's not the case, you have found a call with similar arguments rather than the actual call.

This could be triggered with a call that is with a `closestDiffCount  > 0` and `Repeatability == -1`